gspread/spreadsheet.py

```diff
@@ -437,12 +437,12 @@ def del_worksheet(self, worksheet):
 
         return self.batch_update(body)
 
-    def del_worksheet_by_id(self, worksheet_id: str):
+    def del_worksheet_by_id(self, worksheet_id: int | str):
         """
         Deletes a Worksheet by id
         """
 
-        body = {"requests": [{"deleteSheet": {"sheetId": worksheet_id}}]}
+        body = {"requests": [{"deleteSheet": {"sheetId": str(worksheet_id)}}]}
```

Comment:
you could also extract this beforehand as

```python
worksheet_id_str = str(worksheet_id)
```

just so it is obvious what is happening to it. But this is fine too.